addons/fcs/functions/fnc_keyUp.sqf

```diff
@@ -4,22 +4,27 @@
  * Calculates the offsets for all weapons needed to hit the current target.
  *
  * Arguments:
- * 0: The vehicle
+ * 0: The vehicle <OBJECT>
+ * 1: Turret <CONFIG>
+ * 2: Current Distance <NUMBER>
+ * 3: Show Hint <BOOL>
+ * 4: Play Sound <BOOL>
  *
  * Return Value:
  * none
+ *
+ * Public: No
  */
-
 #include "script_component.hpp"
 
-private ["_vehicle", "_turret", "_turretConfig", "_distance", "_weapons", "_magazines", "_showHint", "_playSound", "_i"];
+private ["_turretConfig", "_weapons", "_magazines", "_showHint", "_playSound", "_i"];
 
-_vehicle = _this select 0;
-_turret = _this select 1;
+params ["_vehicle", "_turret", "_distance", ["_showHint",false], ["_playSound"]];
 
 _turretConfig = [configFile >> "CfgVehicles" >> typeOf _vehicle, _turret] call EFUNC(common,getTurretConfigPath);
-
-_distance = call FUNC(getRange);
+if (isNil "_distance" || {_distance > -1}) then {
```

Comment:
Where is this `_distance > -1` coming from. Why does it have to be greater than -1?